Spec/RestClientChannel.swift

```diff
@@ -155,6 +155,39 @@ class RestClientChannel: QuickSpec {
                     expect(mockExecutor.requests.count).to(equal(1))
                 }
             }
+
+            // RSL1g
+            context("Identified clients with a clientId") {
+
+                // RSL1g1
+                context("when publishing a Message with the clientId attribute set to null") {
+
+                    // RSL1g1a, RSL1g1b
+                    it("should be unnecessary to set clientId of the Message before publishing") {
+                        let options = AblyTests.commonAppSetup()
+                        options.clientId = "john"
+                        let client = ARTRest(options: options)
+                        let channel = client.channels.get("test")
+
+                        waitUntil(timeout: testTimeout) { done in
+                            let message = ARTMessage(name: nil, data: "message")
+                            expect(message.clientId).to(beNil())
+
+                            channel.publish([message]) { error in
+                                expect(error).to(beNil())
+                                channel.history { page, error in
+                                    expect(error).to(beNil())
+                                    expect(page!.items[0].clientId).to(equal("john"))
```

Comment:
This is right, but I think http://docs.ably.io/client-lib-development-guide/features/#RSL1g1b is wrong.  Do you agree @tcard that the spec is wrong?